elasticsearch/src/main/java/org/apache/calcite/adapter/elasticsearch/QueryBuilders.java

```diff
@@ -497,14 +510,80 @@ static class RegexpQueryBuilder extends QueryBuilder {
     private final String fieldName;
     @SuppressWarnings("unused")
     private final String value;
+    @SuppressWarnings("unused")
+    private String escape;
 
     RegexpQueryBuilder(final String fieldName, final String value) {
+      this(fieldName, value, "\\");
+    }
+
+    RegexpQueryBuilder(final String fieldName, final String value, final String escape) {
+      requireNonNull(fieldName, "fieldName");
+      requireNonNull(value, "value");
+      requireNonNull(escape, "escape");
       this.fieldName = fieldName;
-      this.value = value;
+      this.escape = escape;
+      // replace % to * and _ to ? for sql with like operator
+      HashMap<String, String> kv = new HashMap<>();
+      kv.put("%", "*");
+      kv.put("_", "?");
+      this.value = replaceWildcard(value, kv, escape);
+    }
+
+    public static String replaceWildcard(String value, Map<String, String> kv, String escape) {
```

Comment:
SqlFunctions has an implementation of like, I wonder whether anything there can be reused.